File: items-marketplace/src/main/java/me/darkovrbaski/items/marketplace/service/intefaces/OrderService.java
package me.darkovrbaski.items.marketplace.service.intefaces;

import java.util.List;
import me.darkovrbaski.items.marketplace.dto.OrderDto;
import org.springframework.data.domain.Page;

public interface OrderService {

  OrderDto getOrder(Long id);

  OrderDto getUserOrder(Long id, String username);

  List<OrderDto> getActiveOrders(Long userId);

  Page<OrderDto> getHistoryOrders(Long userId, int page, int size);

  void closeOrder(Long id);

  OrderDto trade(OrderDto orderDto, Long matchedOrderId);

  OrderDto createOrder(OrderDto orderDto);

  Page<OrderDto> getMatchedSellOrders(OrderDto orderDto, int page, int size);

}

File: angular-app/src/app/service/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Order, Pagable, Page } from '@app/model';
import { environment } from '@environments/environment.development';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class OrderService {
  private orderUrl = `${environment.apiUrl}/order`;

  constructor(private http: HttpClient) {}

  getOrder(orderId: number): Observable<Order> {
    return this.http.get<Order>(`${this.orderUrl}/${orderId}`);
  }

  getUserOrder(orderId: number): Observable<Order> {
    return this.http.get<Order>(`${this.orderUrl}/user/${orderId}`);
  }

  getActiveOrders(): Observable<Order[]> {
    return this.http.get<Order[]>(`${this.orderUrl}/active`);
  }

  getHistoryOrders(page: Page): Observable<Pagable<Order>> {
    return this.http.get<Pagable<Order>>(
      `${this.orderUrl}/history?page=${page.number}&size=${page.size}`
    );
  }

  createOrder(order: Order): Observable<Order> {
    return this.http.post<Order>(`${this.orderUrl}`, order);
  }

  deleteOrder(orderId: number): Observable<Order> {
    return this.http.delete<Order>(`${this.orderUrl}/${orderId}`);
  }

  getMatchedSellOrders(order: Order, page: Page): Observable<Pagable<Order>> {
    return this.http.post<Pagable<Order>>(
      `${this.orderUrl}/matched?page=${page.number}&size=${page.size}`,
      order
    );
  }

  trade(order: Order, matchedOrderId: number): Observable<Order> {
    return this.http.post<Order>(
      `${this.orderUrl}/trade/${matchedOrderId}`,
      order
    );
  }
}
